File: LeetCode/Python/0485_Max_Consecutive_Ones.py
# 485. Max Consecutive Ones
# https://leetcode.com/problems/max-consecutive-ones/


class Solution:
    def findMaxConsecutiveOnes(self, nums: list[int]) -> int:
        max_cons, temp = 0, 0

        for num in nums:
            if num == 1:
                temp += 1
            else:
                if max_cons < temp:
                    max_cons = temp
                temp = 0

        if max_cons < temp:
            max_cons = temp

        return max_cons

File: LeetCode/JavaScript/0485-Max-Consecutive-Ones.js
// 485. Max Consecutive Ones
// https://leetcode.com/problems/max-consecutive-ones/

/**
 * @param {number[]} nums
 * @return {number}
 */

var findMaxConsecutiveOnes = function (nums) {
  let maxCons = 0,
    temp = 0;

  for (const num of nums) {
    if (num === 1) temp++;
    else temp = 0;
    maxCons = Math.max(maxCons, temp);
  }

  return maxCons;
};
